src/hardware/mixer.cpp

```diff
@@ -707,74 +690,18 @@ static void MIXER_Mix_NoSound()
-#define INDEX_SHIFT_LOCAL 14
+static Bit16s queue_buffer[MIXER_BUFSIZE * MIXER_SSIZE];
 
-static void SDLCALL MIXER_CallBack(MAYBE_UNUSED void *userdata, Uint8 *stream, int len)
+static void MIXER_QueueAudio(Bitu len)
 {
-	memset(stream, 0, len);
-	Bitu need=(Bitu)len/MIXER_SSIZE;
-	Bit16s * output=(Bit16s *)stream;
-	Bitu reduce;
+	Bit16s *output = queue_buffer;
+	Bitu reduce = len;
 	Bitu pos;
-	//Local resampling counter to manipulate the data when sending it off to the callback
-	Bitu index_add = (1<<INDEX_SHIFT_LOCAL);
-	Bitu index = (index_add%need)?need:0;
 
 	Bits sample;
-	/* Enough room in the buffer ? */
-	if (mixer.done < need) {
-//		LOG_MSG("Full underrun need %d, have %d, min %d", need, mixer.done, mixer.min_needed);
-		if((need - mixer.done) > (need >>7) ) //Max 1 percent stretch.
-			return;
-		reduce = mixer.done;
-		index_add = (reduce << INDEX_SHIFT_LOCAL) / need;
-		mixer.tick_add = calc_tickadd(mixer.freq+mixer.min_needed);
-	} else if (mixer.done < mixer.max_needed) {
-		Bitu left = mixer.done - need;
-		if (left < mixer.min_needed) {
-			if( !Mixer_irq_important() ) {
-				Bitu needed = mixer.needed - need;
-				Bitu diff = (mixer.min_needed>needed?mixer.min_needed:needed) - left;
-				mixer.tick_add = calc_tickadd(mixer.freq+(diff*3));
-				left = 0; //No stretching as we compensate with the tick_add value
-			} else {
-				left = (mixer.min_needed - left);
-				left = 1 + (2*left) / mixer.min_needed; //left=1,2,3
-			}
-//			LOG_MSG("needed underrun need %d, have %d, min %d, left %d", need, mixer.done, mixer.min_needed, left);
-			reduce = need - left;
-			index_add = (reduce << INDEX_SHIFT_LOCAL) / need;
-		} else {
-			reduce = need;
-			index_add = (1 << INDEX_SHIFT_LOCAL);
-//			LOG_MSG("regular run need %d, have %d, min %d, left %d", need, mixer.done, mixer.min_needed, left);
-
-			/* Mixer tick value being updated:
-			 * 3 cases:
-			 * 1) A lot too high. >division by 5. but maxed by 2* min to prevent too fast drops.
-			 * 2) A little too high > division by 8
-			 * 3) A little to nothing above the min_needed buffer > go to default value
-			 */
-			Bitu diff = left - mixer.min_needed;
-			if(diff > (mixer.min_needed<<1)) diff = mixer.min_needed<<1;
-			if(diff > (mixer.min_needed>>1))
-				mixer.tick_add = calc_tickadd(mixer.freq-(diff/5));
-			else if (diff > (mixer.min_needed>>2))
-				mixer.tick_add = calc_tickadd(mixer.freq-(diff>>3));
-			else
-				mixer.tick_add = calc_tickadd(mixer.freq);
-		}
-	} else {
-		/* There is way too much data in the buffer */
-//		LOG_MSG("overflow run need %d, have %d, min %d", need, mixer.done, mixer.min_needed);
-		if (mixer.done > MIXER_BUFSIZE)
-			index_add = MIXER_BUFSIZE - 2*mixer.min_needed;
-		else
-			index_add = mixer.done - 2*mixer.min_needed;
-		index_add = (index_add << INDEX_SHIFT_LOCAL) / need;
-		reduce = mixer.done - 2* mixer.min_needed;
-		mixer.tick_add = calc_tickadd(mixer.freq-(mixer.min_needed/5));
-	}
+
+	memset(queue_buffer, 0, sizeof(queue_buffer));
```

Comment:
Suggest std::fill_n() as the c++ way to fill values - and can give it 'len' as the n.